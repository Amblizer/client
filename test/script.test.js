const React = global.window.React = require('react');
const Redux = global.window.Redux = require('redux');
const ReactDOM = global.window.ReactDOM = require('react-dom');
const ReactRedux = global.window.ReactRedux = require('react-redux');
const ReactJSS = global.window.reactJss = require('react-jss');
const io = global.window.io = require('socket.io-client');

const renderer = require('react-test-renderer');

// Mock the Socket.io connect function for all tests
io.connect = jest.fn(() => ({on: jest.fn()}));

const { store, Message, MessageList, Input, Chat } = require('../src/script');

// What do I want to test/make sure works with absolution
// - redux ACTIONS are doing what they intend to do (both sides, firing and
//  components reception)
//    - UI_OPEN
//    - UI_CLOSE
//    - UI_STYLE_MESSANGER
//    - UI_STYLE_MESSANGER
//    - CHAT_CONNECTED
//    - CHAT_DISCONNECTED
//    - CHAT_MESSAGE_OPERATOR
//    - CHAT_MESSAGE_CLIENT
// - styling is available in state
// - ?
//

describe('state', () => {
  it('has a ui.style property', () => {
    const state = store.getState();

    expect(state.ui.hasOwnProperty('style')).toBe(true);

    expect(state.ui.style).not.toBe(undefined);
  });
});

describe('Message', () => {});

describe('MessageList', () => {});

describe('Messages', () => {});

describe('Input', () => {});

describe('Chat', () => {
  it('starts a socket connection', () => {
    const component = renderer.create(
      <Chat store={store} />
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

    expect(io.connect).toHaveBeenCalled();
  })
});
