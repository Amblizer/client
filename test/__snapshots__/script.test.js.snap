exports[`Chat matches snapshot 1`] = `
<div
  className="outerWrapper-446542585">
  <div
    className="box-730084421">
    <div
      className="innerWrapper-2527338721">
      <div
        className="header-89782836">
        <span
          className="headerText-5987815">
          <strong>
            John
          </strong>
           from 
          ACME
        </span>
        <button
          className="icon-1953293147"
          onClick={[Function]}>
          ×
        </button>
      </div>
      <div
        className="messages-3666550760">
        <div
          className="messagesWrapper-901972915">
          <ul
            className="messagesList-539233948" />
        </div>
      </div>
      <textarea
        className="input-2257669887"
        onKeyDown={[Function]}
        placeholder="Type a message…"
        type="text" />
    </div>
  </div>
</div>
`;

exports[`Input matches snapshot 1`] = `
<textarea
  className="input-2257669887"
  onKeyDown={[Function]}
  placeholder="Type a message…"
  type="text" />
`;

exports[`Message matches snapshot 1`] = `
<li
  className="letschat-message"
  id="message_undefined"
  style={
    Object {
      "clear": "both",
    }
  }>
  <div>
    <ul
      className="userContent-2427757672">
      <li>
        Foo Bar
      </li>
    </ul>
  </div>
  <span
    className="letschat-message-timestamp" />
</li>
`;

exports[`MessageList matches snapshot 1`] = `
<div
  className="messages-3666550760">
  <div
    className="messagesWrapper-901972915">
    <ul
      className="messagesList-539233948" />
  </div>
</div>
`;
